test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -274,6 +276,48 @@ public void TempData_CanSetAndGetValues()
             Assert.Equal(input, result);
         }
 
+        public static TheoryData<object, Type> IncompatibleModelData
+        {
+            get
+            {
+                // Small grab bag of instances and expected types with no common base except typeof(object).
```

Comment:
I think what you've done here is fine, I don't know that we need to expose more from the VDD.